### PR TITLE
Better way to find all the dependencies

### PR DESCRIPTION
**Pull request by hcarneiro**
Fixes the following: https://github.com/Fliplet/fliplet-studio/issues/4535

Before this change the code would only find the first level of dependencies, but there are cases that we have a field that depends on another and that other field depends on yet another one.

### CODE REVIEW

File: src/store/index.js [hunk 1]
```diff
@@ -502,50 +502,40 @@ export function sendCssToFrame(value, currentField) {
...
-          const savedValues = getSavedValue(field, true)
+      const savedValues = getSavedValue(field, true)
 
-          if (savedValues.generalSavedValue
-            || savedValues.generalLocalSavedValue
-            || savedValues.widgetSavedValue
-            || savedValues.widgetLocalSavedValue) {
-            return
-          }
+      if (savedValues.generalSavedValue
```
Comment (squallstar) [inline src/store/index.js]: - what is a `general` saved value?
what is a `local` saved value?

File: src/store/index.js [hunk 2]
```diff
@@ -569,6 +559,31 @@ function updateWidgetData(data) {
...
+        variable.fields.forEach((field) => {
+          const inheritingVariable = isInheriting(field.default);
+
+          if (!inheritingVariable || inheritingVariable !== cField.name) {
+            return
+          }
+
+          result.push(field)
+          recursiveFind(field)
+          return
```
Comment (squallstar) [inline src/store/index.js]: what's this return for

File: src/store/index.js (hunk 1 above)
Comment (hcarneiro) [reply src/store/index.js]: @squallstar In the context of everything `general` is referred as the settings for all components. So when it says `widget` that are settings specifically for a widget.

In regards to `local` means settings that are "saved" locally in memory. When it doesn't say `local` are the settings that are already stored on the DB as part of the theme instance settings.

File: src/store/index.js (hunk 2 above)
Comment (hcarneiro) [reply src/store/index.js]: Now that you point it out, I realise it doesn't do anything.

**Pull request merged**